features/steps/advise.py

```diff
@@ -230,8 +230,10 @@ def step_impl(context, runtime_environment: str, user_stack: str, static_analysi
     no_static_analysis = static_analysis == "without"
 
     config._configuration = None  # TODO: substitute with config.reset_config() once new thamos is released
+    config.load_config_from_file(os.path.join(context.repo.working_tree_dir, ".thoth.yaml"))
     config.explicit_host = context.user_api_host
-    with cwd(context.repo.working_tree_dir):
+    working_dir = os.path.join(context.repo.working_tree_dir, config.content.get("overlays_dir"), runtime_environment)
+    with cwd(working_dir):
```

Comment:
Nice catch! 👍🏻 

I thought we have this logic in `advise_here` but it is more low level.

It could be better to do:

```python
    with cwd(configuration.get_overlays_directory(runtime_environment)):
```

as we do in thamos: https://github.com/thoth-station/thamos/blob/8637f79138e315e1a3108802d1800829b2a89a2b/thamos/cli.py#L567

 I wonder if we have this correctly in other parts so the integration correctly picks the right overlays directory.

CC @pacospace @KPostOffice

Reply:
acknowledge, that was my first intuition too, to use `get_overlays_directory(runtime_environment)`
it keeps failing at: https://github.com/thoth-station/thamos/blob/36df7564f5f9cf3e6af07a95a44fab6f90d07a37/thamos/config.py#L676
specifically `os.path.isdir(path)`
with error:
```
        File "/home/hnalla/.local/share/virtualenvs/integration-tests-OOjB5YLZ/lib/python3.8/site-packages/thamos/config.py", line 682, in get_overlays_directory
          raise ConfigurationError(
      thamos.exceptions.ConfigurationError: The directory structure for 'download-dataset' is not initialized yet, you can initialize it by adding packages using `thamos add <pkg> --runtime-environment 'download-dataset'`
```
Reason, as we would be running in the integration-tests dir, whereas the overlays dir is in tmp. 
either we would have to change the dir and then make the request.
so I directly did that here.

Reply:
If I understand correctly, would something like this work here?

```python
with cwd(context.repo.working_tree_dir), cwd(configuration.get_overlays_directory(runtime_environment)):
```

That way we would reuse logic from thamos lib, covering it in these tests.

Reply:
you mean like this:
```
with cwd(context.repo.working_tree_dir): 
  with cwd(config.get_overlays_directory(runtime_environment)):
```

this would work.

Reply:
Made the change. Ready for review.